Firestore/core/test/unit/remote/bloom_filter_test.cc

```
@@ -132,6 +142,131 @@ TEST(BloomFilterTest,
   }
 }
 
+class BloomFilterGoldenTest : public ::testing::Test {
+ public:
+  void RunGoldenTest(std::string& test_file) {
+    size_t start_pos = test_file.find("bloom_filter_proto");
+    if (start_pos == std::string::npos) {
+      return;
+    }
+    std::string result_file = test_file;
+    result_file.replace(start_pos, sizeof("bloom_filter_proto") - 1,
+                        "membership_test_result");
+
+    json test_file_json = ReadFile(test_file);
+    json result_file_json = ReadFile(result_file);
+
+    nlohmann::json bits = reader.OptionalObject("bits", test_file_json, {});
+    std::string bitmap = reader.OptionalString("bitmap", bits, "");
+    int padding = reader.OptionalInt("padding", bits, 0);
+    int hash_count = reader.OptionalInt("hashCount", test_file_json, 0);
+    std::string decoded;
+    absl::Base64Unescape(bitmap, &decoded);
+    std::vector<uint8_t> decoded_map(decoded.begin(), decoded.end());
+    BloomFilter bloom_filter = BloomFilter(decoded_map, padding, hash_count);
+
+    std::string membership_result =
+        reader.OptionalString("membershipTestResults", result_file_json, "");
+
+    for (size_t i = 0; i < membership_result.length(); i++) {
+      bool expectedResult = membership_result[i] == '1';
+      bool mightContainResult = bloom_filter.MightContain(
+          GOLDEN_DOCUMENT_PREFIX_ + std::to_string(i));
+
+      EXPECT_EQ(mightContainResult, expectedResult);
+    }
+  }
+
+ private:
+  std::string GOLDEN_DOCUMENT_PREFIX_ =
+      "projects/project-1/databases/database-1/documents/coll/doc";
+  Path GOLDEN_TEST_FOLDER_ = (Path::FromUtf8(__FILE__).Dirname())
+                                 .AppendUtf8("bloom_filter_golden_test_data/");
+
+  json ReadFile(std::string& file_name) {
+    Path file_path = GOLDEN_TEST_FOLDER_.AppendUtf8(file_name);
+    std::ifstream stream(file_path.native_value());
+    return nlohmann::json::parse(stream);
```

How does `nlohmann::json::parse()` behave if parsing fails?

---

It will throw an exception: https://json.nlohmann.me/features/parsing/parse_exceptions/ 
I will add a try catch block here.

---

No need to add a try/catch block. That just adds noise to the code. Just let the exception propagate and fail the test. I just wanted to make sure that it wasn't silently ignored.